flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/FactoryUtil.java

```diff
@@ -125,12 +125,14 @@
     public static final String PLACEHOLDER_SYMBOL = "#";
 
     /**
-     * Creates a {@link DynamicTableSource} from a {@link CatalogTable}.
+     * Creates a {@link DynamicTableSource}.
      *
-     * <p>It considers {@link Catalog#getFactory()} if provided.
+     * <p>If {@param preferredFactory} is passed, the table source is created from that factory.
+     * Otherwise, an attempt is made to discover a matching factory using Java SPI (see {@link
+     * Factory} for details).
      */
-    public static DynamicTableSource createTableSource(
-            @Nullable Catalog catalog,
+    public static DynamicTableSource createSource(
```

Comment:
I shortened the name here because simply adding an overload (replacing the first argument) would be a breaking change when `null` is passed (as it would cause an ambiguity).